From e8db05ade5aaf827a71231db5d735681d820296f Mon Sep 17 00:00:00 2001
From: Jung Zhao <jung.zhao@rock-chips.com>
Date: Tue, 27 Mar 2018 10:37:59 +0800
Subject: [PATCH] video: rockchip: vpu: add strict cond for vpu_reset

vpu_reset inside try_set_reg must be called after run to done.

Change-Id: Ide396d5c893cfeeea2e6494edc9cd5fd1130974f
Signed-off-by: Jung Zhao <jung.zhao@rock-chips.com>
---
 drivers/video/rockchip/vcodec/vcodec_service.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/rockchip/vcodec/vcodec_service.c b/drivers/video/rockchip/vcodec/vcodec_service.c
index d09699636aa3..9eab291cbd30 100644
--- a/drivers/video/rockchip/vcodec/vcodec_service.c
+++ b/drivers/video/rockchip/vcodec/vcodec_service.c
@@ -1553,6 +1553,9 @@ static void try_set_reg(struct vpu_subdev_data *data)
 {
 	struct vpu_service_info *pservice = data->pservice;
 	int reset_request = atomic_read(&pservice->reset_request);
+	struct vpu_reg *reg_codec = pservice->reg_codec;
+	struct vpu_reg *reg_pproc = pservice->reg_pproc;
+	bool change_able = (!reg_codec) && (!reg_pproc);
 
 	vpu_debug_enter();
 
@@ -1562,7 +1565,7 @@ static void try_set_reg(struct vpu_subdev_data *data)
 		return;
 	}
 
-	if (reset_request) {
+	if (change_able && reset_request) {
 		vpu_service_power_on(data, pservice);
 		mutex_lock(&pservice->reset_lock);
 		vpu_reset(data);
@@ -1570,10 +1573,7 @@ static void try_set_reg(struct vpu_subdev_data *data)
 	}
 
 	if (!list_empty(&pservice->waiting)) {
-		struct vpu_reg *reg_codec = pservice->reg_codec;
-		struct vpu_reg *reg_pproc = pservice->reg_pproc;
 		int can_set = 0;
-		bool change_able = (reg_codec == NULL) && (reg_pproc == NULL);
 		struct vpu_reg *reg = list_entry(pservice->waiting.next,
 				struct vpu_reg, status_link);
 
-- 
2.35.3

